CI: Python 3.14 test failure — changed TypeError message for 'in' operator

## Problem

The Python 3.14 CI job fails with one test failure:

```
FAILED src/hklpy2/tests/test_ops.py::test__fromdict[real_axes-not-iterable]
  AssertionError: Regex pattern did not match.
    Expected regex: "argument\ of\ type\ 'int'\ is\ not\ iterable"
    Actual message: "argument of type 'int' is not a container or iterable"
```

Python 3.14 changed the `TypeError` message when using the `in` operator on a non-iterable type. The message was `"argument of type 'int' is not iterable"` and is now `"argument of type 'int' is not a container or iterable"`.

## Fix

Update the `match` pattern in the test to accept both message variants. For example:

```python
match=re.escape("argument of type 'int' is not")
```

This truncated pattern matches both the old and new message.

## Environment

- Python 3.14 (CI, GitHub Actions)
- All other Python versions (3.10–3.13) pass

Agent: OpenCode (claudeopus46)

--- a/RELEASE_NOTES.rst
+++ b/RELEASE_NOTES.rst
@@ -49,6 +49,14 @@ describe future plans.
       ``extra_axis_names`` are all driven by registered ``GeometryDescriptor``
       objects instead of hard-coded string dispatch. (:issue:`292`, :issue:`293`)
 
+    Fixes
+    -----
+
+    * Fix CI test failure on Python 3.14: ``TypeError`` message for ``in``
+      operator on non-iterable changed from ``"is not iterable"`` to
+      ``"is not a container or iterable"``.  Truncate match pattern to
+      accept both. (:issue:`304`)
+
     Documentation
     -------------
 

--- a/src/hklpy2/tests/test_ops.py
+++ b/src/hklpy2/tests/test_ops.py
@@ -866,9 +866,7 @@ def test_extras_setter(
                 "samples": {},
                 "constraints": {"tth": {"class": "LimitsConstraint", "label": 0}},
             },
-            pytest.raises(
-                TypeError, match=re.escape("argument of type 'int' is not iterable")
-            ),
+            pytest.raises(TypeError, match=re.escape("argument of type 'int' is not")),
             id="real_axes-not-iterable",
         ),
         pytest.param(  # 18

